### PR TITLE
Pill Collector Compatibility

### LINKED ISSUE #2253: Pill collector doesn't allow you to easily put all the pills in one bottle
Or to dump them all in the trash.

What's it even for then?

### PR DESCRIPTION
**Pull request by Kurfursten**
Now compatible with disposal, grinder, and pill bottle.

fixes #2253

### CODE REVIEW

File: code/game/objects/items/weapons/storage/firstaid.dm [hunk 1]
```diff
@@ -150,6 +150,16 @@
...
 
+/obj/item/weapon/storage/pill_bottle/attackby(var/obj/item/I, var/mob/user)
+	if(!I) return
+	if(istype(I, /obj/item/weapon/storage/bag/chem))
+		var/obj/item/weapon/storage/bag/chem/C = I
+		user << "\blue You transfer the contents of the pill collector."
+		for(var/obj/item/O in C.contents)
+			if(can_be_inserted(O))
+				handle_item_insertion(O, 1)
+	return
```
Comment (clusterfack) [inline code/game/objects/items/weapons/storage/firstaid.dm]: return ..()

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 2]
```diff
@@ -1098,6 +1098,22 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 		src.updateUsrDialog()
 		return 0
 
+	//Fill the machine with the pill collector!
+	if(istype(O, /obj/item/weapon/storage/bag/chem))
```
Comment (clusterfack) [inline code/modules/reagents/Chemistry-Machinery.dm]: This is almost entirely copied from the previous section. If they really are so similar you can just use istype(pill collector) || istype(plant bag).

File: code/modules/recycling/disposal.dm [hunk 3]
```diff
@@ -133,6 +133,15 @@
 			update()
 			return
 
+		if(istype(I, /obj/item/weapon/storage/bag/chem))
```
Comment (clusterfack) [inline code/modules/recycling/disposal.dm]: Same as my previous note, if this actually works as is, then you can just have it combined with the trash bag check above.

File: code/game/objects/items/weapons/storage/firstaid.dm [hunk 4]
```diff
@@ -150,6 +150,16 @@
 			return
 	return
 
+/obj/item/weapon/storage/pill_bottle/attackby(var/obj/item/I, var/mob/user)
+	if(!I) return
+	if(istype(I, /obj/item/weapon/storage/bag/chem))
+		var/obj/item/weapon/storage/bag/chem/C = I
+		user << "\blue You transfer the contents of the pill collector."
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/storage/firstaid.dm]: Spans.

File: code/game/objects/items/weapons/storage/firstaid.dm [hunk 5]
```diff
@@ -150,6 +150,16 @@
...
 	return
 
+/obj/item/weapon/storage/pill_bottle/attackby(var/obj/item/I, var/mob/user)
+	if(!I) return
+	if(istype(I, /obj/item/weapon/storage/bag/chem))
+		var/obj/item/weapon/storage/bag/chem/C = I
+		user << "\blue You transfer the contents of the pill collector."
+		for(var/obj/item/O in C.contents)
+			if(can_be_inserted(O))
+				handle_item_insertion(O, 1)
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/storage/firstaid.dm]: Need a return 1 after this loop.

File: code/modules/recycling/disposal.dm [hunk 6]
```diff
@@ -133,6 +133,15 @@
 			update()
 			return
 
+		if(istype(I, /obj/item/weapon/storage/bag/chem))
+			var/obj/item/weapon/storage/bag/chem/C = I
+			user << "\blue You empty the pill collector."
```
Comment (ComicIronic) [inline code/modules/recycling/disposal.dm]: Spans

File: code/modules/recycling/disposal.dm (hunk 3 above)
Comment (ComicIronic) [inline code/modules/recycling/disposal.dm]: You should also add a contents check to this i.e. it should be possible to place the pill dispenser in the disposal should it be empty.

File: code/modules/reagents/Chemistry-Machinery.dm (hunk 2 above)
Comment (Kurfursten) [inline code/modules/reagents/Chemistry-Machinery.dm]: I considered that, but I thought perhaps there was a reason they were casting the item inside the bag as grown.

File: code/modules/recycling/disposal.dm [hunk 7]
```diff
@@ -124,12 +124,16 @@
...
-			var/obj/item/weapon/storage/bag/trash/T = I
-			user << "\blue You empty the bag."
-			for(var/obj/item/O in T.contents)
-				T.remove_from_storage(O,src)
-			T.update_icon()
+		if(istype(I, /obj/item/weapon/storage/bag/))
+			var/obj/item/weapon/storage/bag/B = I
+			if(B.contents.len == 0)
+				user << "<span class='notice'> You throw away the empty [B].</span>"
+				B.loc = src
```
Comment (ComicIronic) [inline code/modules/recycling/disposal.dm]: Can't just do this. You need to use the proper mob handling proc.
`user.drop_item(src)`

**Comment by SkowronX** [comment]
Aw ye. Merge this now @d3athrow

**Pull request merged**